scenarios/command/ts/protoMessages/src/unlockRequestSerializer.ts

```diff
@@ -0,0 +1,19 @@
+import { IMessageSerializer } from '@mqttapplicationsamples/mqttjsclientextensions';
+import {
+    UnlockRequest
+} from './generated/src/proto/unlock_command';
+
+export class UnlockRequestSerializer implements IMessageSerializer {
+    public contentType = "application/protobuf";
+
+    public fromBytes<T>(payload: Buffer): T {
+        return UnlockRequest.decode(payload) as T;
+    }
+
+    public toBytes<T>(payload: T): Buffer {
+        const unlockRequest = UnlockRequest.fromJSON(payload);
```

Comment:
fromJSON? I thought this was proto

Reply:
I could use some collaboration on the generics here. I may indeed be able to use the underlying stream to combine the UnlockRequest and UnlockResponse serializer. But, I was confused about some of the mechanics here.

Reply:
have you tried to use [protobuf.js](https://github.com/protobufjs/protobuf.js/) ? it allows to load the protos at runtime and includes a decoder:

here is an example 
https://github.com/iotmodels/iotux-mqtt/blob/main/telemetryProto.js#L58

Reply:
I had protobuf.js in from the beginning but swapped it out in favor of a better TypeScript generator. Also, I don't think it was feasible to load the proto at runtime since we needed it in both the initial app, a sub-module of the app, and then in some common code. It also didn't compile the .js into .ts it only provided the .ts types. This required a bunch of pre-post build custom steps. All of that might be worth not having to pre-install the proto compiler. Let's chat about this.